src/seraphis_mocks/mock_http_client_pool.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// NOT FOR PRODUCTION
+
+//local headers
+#include "mock_http_client_pool.h"
+
+//third party headers
+
+//standard headers
+
+//forward declarations
+
+
+namespace sp
+{
+namespace mocks
+{
+//-------------------------------------------------------------------------------------------------------------------
+std::size_t ClientConnectionPool::acquire_unused_http_client()
+{
+    std::lock_guard<std::mutex> lock{m_http_client_pool_mutex};
+
+    std::size_t http_client_index_out = 0;
+
+    // TODO: if m_http_client_pool.size() >= m_max_connections, wait until unused
+
+    // Check if we have any clients available for use
+    bool found_unused_client = false;
+    for (std::size_t i = 0; i < m_http_client_pool.size(); ++i)
+    {
+        if (!m_http_client_pool[i].in_use)
+        {
+            http_client_index_out = i;
+            found_unused_client = true;
+            break;
+        }
+    }
```

Comment:
```suggestion
    // Check if we have any clients available for use
    for (std::size_t i = 0; i < m_http_client_pool.size(); ++i)
    {
        if (m_http_client_pool[i].in_use)
            continue;

        m_http_client_pool[i].in_use = true;
        return i;
    }
```
Simpler, less nesting here and after. I am a *huge* fan of early-out/early-continue patterns like this.